### PR TITLE
Refactor camera permission

### LINKED ISSUE #260: Enhance camera request permission
Currently, the user needs to click a button to initiate the permission request, as demonstrated in this video:  

#TODO: ADD VIDEO  

Based on user feedback, it would be preferable for the permission request to be triggered automatically when the user interacts with a feature requiring the camera (e.g., scanning an ingredient or taking a picture).

- [ ] Refactor `RequestCameraPermission.kt`

### PR DESCRIPTION
**Pull request by Ylanv**
### What has been changed?

This PR improves the way camera permissions are handled within the app. The changes include:
- A smoother permission request flow.
- Clear explanations to users about why the permission is needed.
- Instructions on how to enable permissions in the app settings if previously denied.

### Why was this change made?

These changes enhance the user experience by ensuring a more intuitive and informative permission request process. By providing users with clear guidance, we aim to reduce confusion and improve app usability.

### Video

https://github.com/user-attachments/assets/580285e2-e296-4808-b4db-77dcea2497a3

### Checklist
- [x] Tests have been added to cover the changes.
- [x] Documentation has been updated to reflect the new behavior.
- [x] All CI checks have passed successfully.
- [x] Linked issue/feature: #260

### CODE REVIEW

**Review by 2Fick — CHANGES_REQUESTED** [review]
Fantastic work! You’ve handled permissions like a 🐱‍💻 Hacker Cat—clean, precise, and efficient. The test coverage is solid, and the updated `FakePermissionState` feels as versatile as a 🐱‍🚀 Astro Cat, ready to tackle all edge cases in the permission universe. 🚀

I can see your attention to detail is on par with a 🐱‍👓 Hipster Cat—well-organized strings and clean code structure are truly a style statement. But, let’s not forget about the 'Back' button suggestion! Give users a smooth exit, and when you do, test it thoroughly, or the boss might go full 🐱‍🐉 Dino Cat mode! 😅

Overall, you’re coding through tasks like a 🐱‍🏍 Stunt Cat—daring, focused, and sticking the landing perfectly. Keep it up! 👏

File: app/src/androidTest/java/com/android/sample/camera/CameraPermissionTest.kt [hunk 1]
```diff
@@ -24,7 +25,9 @@ class CameraPermissionTest {
   @Test
```
[the remark this answers is not in the document]
Comment (Ylanv) [reply app/src/androidTest/java/com/android/sample/camera/CameraPermissionTest.kt:37]: Thank you for pointing that out, great catch! I've added the scaffold to ensure we have a functional go-back button. Let me know if there's anything else I might have overlooked!

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -185,4 +185,7 @@
     <string name="you_are_offline">You are Offline!</string>
     <string name="this_feature_requires_an_internet_connection_please_check_your_connection_and_try_again">This feature requires an internet connection. Please check your connection and try again.</string>
     <string name="offline">Offline</string>
+    <string name="camera_permission_is_required_to_use_this_feature">Camera permission is required to use this feature.</string>
```
[the remark this answers is not in the document]
Comment (Ylanv) [reply app/src/main/res/values/strings.xml:188]: Thanks!

File: app/src/main/java/com/android/sample/feature/camera/RequestCameraPermission.kt [hunk 3]
```diff
@@ -28,20 +37,35 @@ fun RequestCameraPermission(
...
-    is PermissionStatus.Denied -> {
-      Box(contentAlignment = Alignment.Center, modifier = Modifier.fillMaxSize()) {
-        Button(
-            onClick = { permissionState.launchPermissionRequest() },
-            modifier = Modifier.fillMaxWidth(1f / 2f).fillMaxHeight(1f / 10f)) {
-              Text("Request permission")
-            }
-      }
-    }
+
```
[the remark this answers is not in the document]
Comment (Ylanv) [reply app/src/main/java/com/android/sample/feature/camera/RequestCameraPermission.kt:43]: Good one! I've made the modification, thanks for the laugh and the reminder. Let me know if there's anything else to tweak!

**Review by 2Fick — APPROVED** [review]
Awesome work! The back button addition is spot-on and ensures a smooth, user-friendly flo, no more feeling locked in! 🏃‍♂️🔙 Clean implementation and solid testing as always. Great job delivering this update perfectly! 🚀👏

PR approved (97.4% of line coverage)!!
